src/DAQModule.cpp

```diff
@@ -0,0 +1,36 @@
+/**
+ * @file DAQModule.cpp DAQModule class implementation
+ *
+ * This is part of the DUNE DAQ Application Framework, copyright 2020.
+ * Licensing/copyright details are in the COPYING file that you should have
+ * received with this code.
+ */
+
+#include "appfwk/DAQModule.hpp"
+
+namespace dunedaq::appfwk {
+
+void 
+DAQModule::execute_command(const std::string& name, const std::vector<std::string>& args) {
+   if (auto cmd = commands_.find(name); cmd != commands_.end()) {
+      cmd->second(args);
+      return;
+    }
+
+    throw UnknownCommand(ERS_HERE, name);
+}
+
+std::vector<std::string> 
+DAQModule::get_commands() const {
+    std::vector<std::string> cmds(commands_.size());
+    for ( const auto &[key, value] : commands_)
```

Comment:
In case the commads_ 's type is changed, I sussgest
```C+++
for ( const auto & it : command_ ) 
  cmds.push_back( it -> first ) 
```

Reply:
How would that make a difference? Value's type is inferred and its' not used anyway.

Reply:
It's not a problem of the type. It's just that if the type changes the loop can be left unchanged.

Reply:
The loop can be left unchanged only if `it` is an iterator exposing `->first`, which practically means a map iterator under any realistic scenario. Under the same scenario, the structured binding remains valid and it is more readable.